catalog/plugin/catalog-plugin-videothumbnail/src/main/java/org/codice/ddf/catalog/content/plugin/video/VideoThumbnailPlugin.java

```diff
@@ -150,44 +154,53 @@ private void copyFFmpegBinary(final InputStream inputStream) throws IOException
+        processContentItems(input.getCreatedContentItems(), input.getProperties());
         return input;
     }
 
     @Override
     public UpdateStorageResponse process(final UpdateStorageResponse input)
             throws PluginExecutionException {
         // TODO: How to handle application/octet-stream?
-        try {
-            processContentItems(input.getUpdatedContentItems(), input.getProperties());
-        } catch (IllegalArgumentException e) {
-            input.getProcessingErrors()
-                    .add(new ProcessingDetailsImpl(this.getClass()
-                            .getName(), e));
-        }
+        processContentItems(input.getUpdatedContentItems(), input.getProperties());
         return input;
     }
 
     private void processContentItems(final List<ContentItem> contentItems,
-            final Map<String, Serializable> properties)
-            throws PluginExecutionException, IllegalArgumentException {
+            final Map<String, Serializable> properties) throws PluginExecutionException {
         Map<String, Map<String, Path>> tmpContentPaths =
                 (Map<String, Map<String, Path>>) properties.get(Constants.CONTENT_PATHS);
 
         for (ContentItem contentItem : contentItems) {
             if (isVideo(contentItem)) {
-                Map<String, Path> contentPaths = tmpContentPaths.get(contentItem.getId());
 
+                final long contentItemSizeBytes;
+                try {
+                    contentItemSizeBytes = contentItem.getSize();
+                } catch (IOException e) {
+                    LOGGER.warn(
+                            "Error retrieving size for ContentItem (id={}). Skipping creating thumbnail for metacard (id={})");
+                    break;
+                }
+
+                final long bytesPerMegabyte = 1024L * 1024L;
+                final long maxFileSizeBytes = maxFileSizeMB * bytesPerMegabyte;
+                if (contentItemSizeBytes > maxFileSizeBytes) {
+                    LOGGER.debug(
+                            "ContentItem (id={} size={} MB) is larger than the configured max file size to process ({} MB). Skipping creating thumbnail for metacard (id={})",
+                            contentItem.getId(),
+                            contentItemSizeBytes,
+                            maxFileSizeBytes,
+                            contentItem.getMetacard()
+                                    .getId());
+                    break;
+                }
+
+
+                Map<String, Path> contentPaths = tmpContentPaths.get(contentItem.getId());
                 if (contentPaths == null || contentPaths.isEmpty()) {
-                    throw new IllegalArgumentException(
-                            "No path for contentItem " + contentItem.getId()
-                                    + " provided. Skipping.");
+                    LOGGER.warn("No path for ContentItem (id={}). Skipping creating thumbnail");
+                    break;
```

Comment:
Do we want to continue here as well? I realize we threw an exception before but continuing seems like a better option.